client/foundries.go

```diff
@@ -1193,3 +1214,89 @@ func (a *Api) FactoryPatchCA(factory string, certs CaCerts) error {
 	_, err = a.Patch(url, data)
 	return err
 }
+
+func (a *Api) FactoryCreateWave(factory string, wave *WaveCreate) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/"
+	logrus.Debugf("Creating factory wave %s", url)
+
+	data, err := json.Marshal(wave)
+	if err != nil {
+		return err
+	}
+
+	_, err = a.Post(url, data)
+	return err
+}
+
+func (a *Api) FactoryListWaves(factory string) ([]Wave, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/"
+	logrus.Debugf("Listing factory waves %s", url)
+
+	body, err := a.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp []Wave
+	err = json.Unmarshal(*body, &resp)
+	return resp, err
+}
+
+func (a *Api) FactoryGetWave(factory string, wave string, showTargets bool) (*Wave, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/" + wave + "/"
+	if showTargets {
+		url += "?show-targets=1"
+	}
+	logrus.Debugf("Fetching factory wave %s", url)
+
+	body, err := a.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp Wave
+	err = json.Unmarshal(*body, &resp)
+	return &resp, err
+}
+
+func (a *Api) FactoryDiscardWave(factory string, wave string) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/" + wave + "/discard/"
+	logrus.Debugf("Discarding factory wave %s", url)
+	_, err := a.Post(url, nil)
+	return err
+}
+
+func (a *Api) FactoryPromoteWave(factory string, wave string) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/" + wave + "/promote/"
+	logrus.Debugf("Promoting factory wave %s", url)
+	_, err := a.Post(url, nil)
+	return err
+}
+
+func (a *Api) PromotedTargetsList(factory string, tags ...string) (map[string]tuf.SignedTargets, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/promoted-targets/?tag=" + strings.Join(tags, ",")
```

Comment:
We could probably use the current targets API and do `<factory>/targets/?tag=<tag>&production=1`.

Reply:
A current Targets API returns a list of individual CI targets, whereas this API returns a map of production "targets.json" files i.e. they are not compatible.

Reply:
Do we really need to return this map? We are going to be based off a single tag. Feels like we are trying to optimize for a scenario we don't want to encourage/support?

Reply:
You would need this API when doing an offline targets key rotation.  In that case we need to re-sign all production targets and it's best to fetch all of them in a single shot.
I don't want to create two separate APIs for a wave versus key rotation use cases where a single API is easier to implement.
On client side a trivial `PromotedTargetsGet` function adapts that API to a single tag.

Reply:
I'm just getting concerned about API confusion. We already have two APIs for getting targets. Now we are adding a 3rd API because it might be a little quicker on the client side of a command that might get run once a month?

This also affects a command like `fioctl targets list` which would need a totally separate code path for the `--prod` option.

I do realize on the backend its roughly the same code so I might be overthinking.